File: extract-content.js
/**
 * Script to extract language content from musics.csv into JSON files
 * Run with: node extract-content.js
 */

const fs = require('fs');
const Papa = require('papaparse');

// Read musics.csv
const csvData = fs.readFileSync('data/musics.csv', 'utf8');

Papa.parse(csvData, {
    header: true,
    skipEmptyLines: true,
    complete: function(results) {
        const simplifiedRows = [];

        results.data.forEach((row, index) => {
            // Create JSON content file for this music
            const content = {
                country_description: {
                    pt_br: row.country_description_pt_br || '',
                    es_es: row.country_description_es_es || '',
                    en_us: row.country_description_en_us || ''
                },
                song_description: {
                    pt_br: row.song_description_pt_br || '',
                    es_es: row.song_description_es_es || '',
                    en_us: row.song_description_en_us || ''
                },
                teacher_bio: {
                    pt_br: row.teacher_bio_pt_br || '',
                    es_es: row.teacher_bio_es_es || '',
                    en_us: row.teacher_bio_en_us || ''
                }
            };

            // Write JSON file
            fs.writeFileSync(
                `content/musics/${row.id}.json`,
                JSON.stringify(content, null, 2)
            );

            // Create simplified row for new CSV
            simplifiedRows.push({
                id: row.id,
                lat: row.lat,
                lng: row.lng,
                country_name: row.country_name,
                country_flag: row.country_flag,
                song_name: row.song_name,
                song_author: row.song_author,
                audio: row.audio,
                teacher_name: row.teacher_name,
                teacher_photo: row.teacher_photo,
                teacher_link: row.teacher_link
            });
        });

        // Create simplified musics.csv
        const newCsv = Papa.unparse(simplifiedRows);
        fs.writeFileSync('data/musics.csv', newCsv);

        console.log(`✅ Extracted content for ${results.data.length} musics`);
        console.log(`✅ Created simplified musics.csv`);
    }
});

File: extract-content.py
#!/usr/bin/env python3
"""
Script to extract language content from musics.csv into JSON files
Run with: python3 extract-content.py
"""

import csv
import json
import os

# Read musics.csv
with open('data/musics.csv', 'r', encoding='utf-8') as f:
    reader = csv.DictReader(f)
    rows = list(reader)

simplified_rows = []

for row in rows:
    # Create JSON content file for this music
    content = {
        "country_description": {
            "pt_br": row.get('country_description_pt_br', ''),
            "es_es": row.get('country_description_es_es', ''),
            "en_us": row.get('country_description_en_us', '')
        },
        "song_description": {
            "pt_br": row.get('song_description_pt_br', ''),
            "es_es": row.get('song_description_es_es', ''),
            "en_us": row.get('song_description_en_us', '')
        },
        "teacher_bio": {
            "pt_br": row.get('teacher_bio_pt_br', ''),
            "es_es": row.get('teacher_bio_es_es', ''),
            "en_us": row.get('teacher_bio_en_us', '')
        }
    }

    # Write JSON file
    with open(f"content/musics/{row['id']}.json", 'w', encoding='utf-8') as json_file:
        json.dump(content, json_file, ensure_ascii=False, indent=2)

    # Create simplified row for new CSV
    simplified_rows.append({
        'id': row['id'],
        'lat': row['lat'],
        'lng': row['lng'],
        'country_name': row['country_name'],
        'country_flag': row['country_flag'],
        'song_name': row['song_name'],
        'song_author': row['song_author'],
        'audio': row['audio'],
        'teacher_name': row['teacher_name'],
        'teacher_photo': row['teacher_photo'],
        'teacher_link': row.get('teacher_link', '')
    })

# Create simplified musics.csv
with open('data/musics.csv', 'w', encoding='utf-8', newline='') as f:
    if simplified_rows:
        writer = csv.DictWriter(f, fieldnames=simplified_rows[0].keys())
        writer.writeheader()
        writer.writerows(simplified_rows)

print(f"✅ Extracted content for {len(rows)} musics")
print(f"✅ Created simplified musics.csv")
